clang/lib/CodeGen/ModuleBuilder.cpp

```diff
@@ -282,6 +288,8 @@ namespace {
     }
 
     void HandleTranslationUnit(ASTContext &Ctx) override {
+      IRGenFinished = true;
```

Comment:
shouldn't this be set at the end of `HandleTranslationUnit` rather than the beginning?

Reply:
Good point. I guess I was avoiding this early return which I have now removed. In fact this change fixed some of the failures we see in the tests.